src/Application.vala

```diff
@@ -202,12 +202,38 @@ public class Terminal.Application : Gtk.Application {
                 notification.set_icon (process_icon);
                 notification.set_default_action_and_target_value ("app.process-finished", new Variant.string (id));
                 send_notification ("process-finished-%s".printf (id), notification);
+
+                ulong tab_change_handler = 0;
+                ulong focus_in_handler = 0;
+
+                tab_change_handler = terminal.main_window.notify["current-terminal"].connect (() => {
+                    withdraw_notification_for_terminal (terminal, id, tab_change_handler, focus_in_handler);
+                });
+
+                focus_in_handler = terminal.main_window.focus_in_event.connect (() => {
+                    withdraw_notification_for_terminal (terminal, id, tab_change_handler, focus_in_handler);
+
+                    return Gdk.EVENT_PROPAGATE;
+                });
+
             }
         });
 
         return true;
     }
 
+    private void withdraw_notification_for_terminal (TerminalWidget terminal, string id, ulong tab_change_handler, ulong focus_in_handler) {
+        if (terminal.main_window.current_terminal != terminal) {
+            return;
+        }
+
+        terminal.tab.icon = null;
+        withdraw_notification ("process-finished-%s".printf (id));
+
+        terminal.main_window.disconnect (tab_change_handler);
+        terminal.main_window.disconnect (focus_in_handler);
```

Comment:
Adding 
```
        if (terminal == terminal.main_window.current_terminal) {
            terminal.tab.icon = null;
        }
```
fixes the icon issue.

Reply:
I already have that on line #230?

Reply:
Sorry, I must have got confused about which branch I was running at some point :disappointed: Your PR does indeed fix that issue!